staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -623,10 +652,62 @@ func preprocessDeletionListForMerging(key string, original map[string]interface{
 	return false, false, "", nil
 }
 
+// processReplaceKeys processes replaceKeys list.
+// When MergeParallelList is true, we validate the patch and
+// then clear the fields that are not in the list.
+func processReplaceKeys(original, patch map[string]interface{}, options MergeOptions) error {
+	replaceKeysInPatch, foundInPatch := patch[replaceKeysDirective]
+	delete(patch, replaceKeysDirective)
+	if !options.MergeParallelList {
+		replaceKeysInOriginal, foundInOriginal := original[replaceKeysDirective]
+		switch {
+		case foundInOriginal && foundInPatch:
+			if !reflect.DeepEqual(replaceKeysInOriginal, replaceKeysInPatch) {
+				return fmt.Errorf("%v and %v are not deep equal", replaceKeysInOriginal, replaceKeysInPatch)
+			}
+		case !foundInOriginal && foundInPatch:
```

Comment:
Both of these switch cases depend on `foundInPatch == true`. In my opinion, this should be refactored as follows:

```go
if (!foundInPatch) {
    return nil
} else if (foundInOriginal) {
    if reflect.DeepEqual(...) { ... }
} else {
    original[replaceKeysDirective] = replaceKeysInPatch
}
```

I think there are even more simplifications possible at the beginning of this function.